### PR TITLE
Update Bitbucket driver to reflect API changes

### PR DESCRIPTION
**Pull request by christianruhstaller**
See Discussion: https://github.com/drone/go-scm/issues/28

### CODE REVIEW

File: scm/driver/bitbucket/git.go [hunk 1]
```diff
@@ -207,14 +207,12 @@ func convertCommit(from *commit) *scm.Commit {
 			Name:   from.Author.User.DisplayName,
 			Email:  extractEmail(from.Author.Raw),
 			Date:   from.Date,
-			Login:  from.Author.User.Username,
```
Comment (tboerger) [inline scm/driver/bitbucket/git.go]: What happened to these fields?

Comment (christianruhstaller) [reply scm/driver/bitbucket/git.go]: Bitbucket changed their api and removed the "login" field:
https://developer.atlassian.com/cloud/bitbucket/bitbucket-api-changes-gdpr/#removal-of-usernames-from-user-referencing-apis

**Comment by tboerger** [comment]
Nickname is not a proper replacement as it doesn't have to be unique.

**Comment by christianruhstaller** [comment]
If the Username needs to be unique. The Bitbucket API won't provide this information anymore. Down the road this means you cannot use the username anymore for usage as an information for plugins and co.

This breaks already: https://discourse.drone.io/t/branch-author-not-works-in-slack-plugin/4624

If the Username can't be used by the Bitbucket driver, drone needs to change its behavior to give the plugins a chance to use something like an username (Like the nickname or display name).

**Comment by matthewhartstonge** [comment]
@tboerger if we want uniqueness, the best way forward would be to use the new Atlassian `account_id` field. Just did a poll of the API and got the following back for a commit:

```
"author": {
	"raw": "Matthew Hartstonge <matt@MY_DOMAIN.com>",
	"type": "author",
	"user": {
		"display_name": "Matthew Hartstonge",
		"uuid": "{A-COOL-DASH-SEPARATED-UUID}",
		"links": {
			"self": {
				"href": "https://api.bitbucket.org/2.0/users/%7BA-COOL-DASH-SEPARATED-UUID%7D"
			},
			"html": {
				"href": "https://bitbucket.org/%7BA-COOL-DASH-SEPARATED-UUID%7D/"
			},
			"avatar": {
				"href": "https://secure.gravatar.com/avatar/1414dfb6e23f3c4b2fab78cb84c3f070?d=https%3A%2F%2Favatar-management--avatars.us-west-2.prod.public.atl-paas.net%2Finitials%2FMH-4.png"
			}
		},
		"nickname": "Matthew Hartstonge",
		"type": "user",
		"account_id": "NUMBER:MY-SECR3T-ACCOUNT-ID"
	}
},

```

I'm happy to help out where I can, what would be the upstream implications/places it would need to be fixed?

**Comment by tboerger** [comment]
That's the point where @bradrydzewski should jump in :)

**Comment by bradrydzewski** [comment]
my thoughts are that the `username` field _must_ store a username. If Bitbucket no longer returns a username then this field will remain blank since no username is available. If we want to make another field available (e.g. nickname) we should add a new field.

**Comment by matthewhartstonge** [comment]
@christianruhstaller you keen to continue with this PR?

**Comment by christianruhstaller** [comment]
I removed the username field, as this does not exist in the Bitbucket API anymore and added the Nickname field which is not used currently.

**Pull request closed without merging**